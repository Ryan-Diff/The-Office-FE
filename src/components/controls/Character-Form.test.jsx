import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterForm from './Character-Form';

describe('Character form component', () => {
    afterEach(() => cleanup());
    it('renders CharacterForm', () => {
        const { asFragment } = render(
            <CharacterForm
              name= 'Dwight'
              position= 'Assistant to the Regional Manager'
              quote= 'Identity theft is not a joke Jim!'
              imageUrl= 'https://upload.wikimedia.org/wikipedia/en/c/cd/Dwight_Schrute.jpg'
              favoriteDessert= 'Can of Pickled Beets from his survivor shelter'
              onChange={() =>{}}
              onSubmit={() =>{}}
              />);
              expect(asFragment()).toMatchSnapshot();
    });
});