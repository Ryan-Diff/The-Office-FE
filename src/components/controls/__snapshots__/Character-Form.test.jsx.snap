// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character form component renders CharacterForm 1`] = `
<DocumentFragment>
  <div>
    <form
      class="postCharacter"
    >
      <label
        for="name"
      >
        Name
      </label>
      <input
        id="name"
        name="name"
        type="text"
        value="Dwight"
      />
      <label
        for="position"
      >
        position
      </label>
      <input
        id="position"
        name="position"
        type="text"
        value="Assistant to the Regional Manager"
      />
      <label
        for="quote"
      >
        Quote
      </label>
      <input
        id="quote"
        name="quote"
        type="text"
        value="Identity theft is not a joke Jim!"
      />
      <label
        for="imageUrl"
      >
        Image Url
      </label>
      <input
        id="imageUrl"
        name="imageUrl"
        type="text"
        value="https://upload.wikimedia.org/wikipedia/en/c/cd/Dwight_Schrute.jpg"
      />
      <label
        for="favoriteDessert"
      >
        Favorite Dessert
      </label>
      <input
        id="favoriteDessert"
        name="favoriteDessert"
        type="text"
        value="Can of Pickled Beets from his survivor shelter"
      />
      <button
        data-testid="formButton"
      >
        Add Character
      </button>
    </form>
  </div>
</DocumentFragment>
`;
